### PR TITLE
Pass thru SET_NOMOCK_FLAG so we can control 'mock' behavior.

### PR DESCRIPTION
**Pull request by david-mcmahon**
cc @ixdy @spxtr

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/345)

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml [hunk 1]
```diff
@@ -47,16 +47,17 @@
...
         - 'federation-build':
             branch: 'master'
             timeout: 50
             job-env: |
               export RELEASE_INFRA_PUSH=true
+              export SET_NOMOCK_FLAG=false
               export PROJECT="k8s-jkns-e2e-gce-federation"
               export FEDERATION=true
-              export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-gce-federation"
+              export FEDERATION_PUSH_REPO_BASE="gcr.io/djmm-federation-build"
```
Comment (nikhiljindal) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: I suggested this assuming that the PROJECT in your job is also same.
Is this the project in which you want to run your job? (Make sure that it has right service accounts)
If yes, then set the PROJECT to "djmm-federation-build" as well.

Comment (david-mcmahon) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: Oh right, of course PROJECT should be synced there, but yes there is no such djmm-federation-build project.  I guess I was asking if there's a FEDERATION_PUSH_REPO_BASE that is used in jenkins already that has everything set up.  Maybe there is not.  I don't see anything.

Maybe it's just fine to push things up to `k8s-jkns-e2e-gce-federation` from this test job?  @spxtr @ixdy ?

Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: That should work.

Comment (david-mcmahon) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: Thanks, @spxtr .  Updated title.  LGTY?

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml [hunk 2]
```diff
@@ -47,16 +47,17 @@
...
             timeout: 50
             job-env: |
               export RELEASE_INFRA_PUSH=true
+              export SET_NOMOCK_FLAG=false
               export PROJECT="k8s-jkns-e2e-gce-federation"
               export FEDERATION=true
               export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-gce-federation"
               export KUBE_FASTBUILD=true
               export KUBE_GCS_RELEASE_BUCKET=kubernetes-federation-release
-              export KUBE_GCS_RELEASE_BUCKET_MIRROR=kubernetes-federation-release
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: was this intentionally removed?

Comment (david-mcmahon) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: Yes.  It looks like its essentially a placeholder when BUCKET==BUCKET_MIRROR, so just removing clutter from this test job.  Correct me if I'm wrong.

Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: I think `KUBE_GCS_RELEASE_BUCKET_MIRROR` had a default set at one point, so we needed to explicitly set it. Not sure if it's still necessary.

Comment (david-mcmahon) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-djmm.yaml]: It's actually not used at all anymore in `hack/jenkins/build.sh` or the new `push-ci-build.sh` for the purposes of this test job.

**Comment by ixdy** [comment]
LGTM

**Pull request merged**